milabench/validation/ensure_rate.py

```diff
@@ -0,0 +1,42 @@
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import List
+
+from .validation import ValidationLayer, BenchLogEntry
+
+
+class _Layer(ValidationLayer):
+    """Makes sure the training rate is generated for each benchmarks"""
+
+    def __init__(self, **kwargs) -> None:
+        self.rates = defaultdict(float)
+        self.errors = 0
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *args):
+        pass
+
+    def on_event(self, entry: BenchLogEntry):
+        tag = entry.tag
+
+        if tag not in self.rates:
+            self.rates[tag] = 0
+
+        if entry.event == "data":
```

Comment:
I feel like we could reduce boilerplate a bit by having the `ValidationLayer` class dispatch to `getattr(self, f"on_{entry.event}")`. Just a thought.

Reply:
done